eth/catalyst/api_test.go

```diff
@@ -1240,9 +1240,10 @@ func TestNilWithdrawals(t *testing.T) {
 
 func setupBodies(t *testing.T) (*node.Node, *eth.Ethereum, []*types.Block) {
 	genesis, blocks := generateMergeChain(10, true)
-	n, ethservice := startEthService(t, genesis, blocks)
 	// enable shanghai on the last block
-	ethservice.BlockChain().Config().ShanghaiTime = &blocks[len(blocks)-1].Header().Time
+	time := blocks[len(blocks)-1].Header().Time + 1
```

Comment:
This does something more than just fix a races, it also bumps the time by `1` second. So it also either fixes or introduces a bug... ?

Reply:
It fixes a bug. Previously shanghai was enabled `before` the last block, now its enabled afterwards. This means previously we would import the block on pre-shanghai rules, then afterwards set the chain config so the last block is post-shanghai. Basically we imported the block and modified the rules afterwards.